### PR TITLE
Support Gurobi 9.5.x

### PR DESCRIPTION
**Pull request by mwoehlke-kitware**
Remove logic that limits Gurobi to 9.5.1, in order to support 9.5.x at any patch version. Update documentation accordingly.

Towards #14157.

- - -
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/18464)

### CODE REVIEW

**Review by mwoehlke-kitware — COMMENTED** [review]
+@jwnimmer-tri for feature review, please.

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/18464#-:-NJLZoEe9aV-F6eXfvZC:b-qf68jg)* status: 1 unresolved discussion, LGTM missing from assignee jwnimmer-tri(platform), needs at least two assigned reviewers, missing label for release notes (waiting on @jwnimmer-tri)

___
*[`doc/_pages/bazel.md` line 183 at r1](https://reviewable.io/reviews/robotlocomotion/drake/18464#-NJLZrLdCjrMdsgIgd2v:-NJLZrLdCjrMdsgIgd2w:ba70vh6) ([raw file](https://github.com/robotlocomotion/drake/blob/fea2405f39a6e28462cbab41cdeb8df18da710d3/doc/_pages/bazel.md#L183)):*
> ```Markdown
> 2. Set up your Gurobi license file in accordance with Gurobi documentation.
> 3. ``export GRB_LICENSE_FILE=/path/to/gurobi.lic``.
> 4. Download ``gurobi9.5.1_linux64.tar.gz``. You may need to manually edit the URL to get the correct version.
> ```

BTW, we still mention "9.5.1" here. I didn't have any good ideas how to reword this to say "any 9.5.x version" without making it more obtuse. Suggestions? (Is it okay as-is?)

**Review by jwnimmer-tri — COMMENTED** [review]
:lgtm: feature, modulo open discussions.

+(release notes: feature)

> Reviewed 6 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/18464#-:-NJW2Syu8l7hi8f57VoC:bcu9apm)* status: 4 unresolved discussions, needs at least two assigned reviewers (waiting on @mwoehlke-kitware)

*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake/18464#-NJW3fbg2pyvq_pJEFql:-NJW3fbg2pyvq_pJEFqm:bw579t5) (no related file):*
The file `drake/cmake/modules/FindGurobi.cmake` looks like it should have been updated as well with revised logic.

I think you need to prepare a suite of CI runs that _only_ have 9.5.2 installed, and prove that this PR passes all of those.

___
___
*[`doc/_pages/bazel.md` line 183 at r1](https://reviewable.io/reviews/robotlocomotion/drake/18464#-NJLZrLdCjrMdsgIgd2v:-NJW18Db6OqJ6n7Q20SJ:bkputjj) ([raw file](https://github.com/robotlocomotion/drake/blob/fea2405f39a6e28462cbab41cdeb8df18da710d3/doc/_pages/bazel.md#L183)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

BTW, we still mention "9.5.1" here. I didn't have any good ideas how to reword this to say "any 9.5.x version" without making it more obtuse. Suggestions? (Is it okay as-is?)
</blockquote></details>

Clarity for users is the most important; we can sacrifice compactness if we need to.

I think we should:

(1) explicitly say in this section of the documentation that Drake supports any `9.5.x` patch version, and
(2) explicitly say that if the user has a different patch version that shown here, to just update all of the paths in these instructions to align with their patch version, and
(3) switch our canonical example here to be 9.5.2, since that's what users are currently being offered by the Gurobi Download Center (also for macOS on line 192 below).
___
*[`tools/workspace/gurobi/package-ubuntu.BUILD.bazel.in` line 28 at r1](https://reviewable.io/reviews/robotlocomotion/drake/18464#-NJW29pv6yVwlJJkd-HZ:-NJW29pv6yVwlJJkd-H_:b4h4s1u) ([raw file](https://github.com/robotlocomotion/drake/blob/fea2405f39a6e28462cbab41cdeb8df18da710d3/tools/workspace/gurobi/package-ubuntu.BUILD.bazel.in#L28)):*
> ```
> ]) or [":error-message.h"]
> 
> # In the Gurobi package, libgurobi95.so is a symlink to libgurobi.so.9.0.0.
> ```

typo

Ditto on the next line as well.



_Suggestion:_
```
9.5.1
```
___
*[`tools/workspace/gurobi/repository.bzl` line 17 at r1](https://reviewable.io/reviews/robotlocomotion/drake/18464#-NJVz_T21gyrsb6WSBio:-NJVz_T21gyrsb6WSBip:b-xi7prs) ([raw file](https://github.com/robotlocomotion/drake/blob/fea2405f39a6e28462cbab41cdeb8df18da710d3/tools/workspace/gurobi/repository.bzl#L17)):*
> ```Starlark
>     if os_result.is_macos:
>         # Gurobi must be installed into its standard location.
>         gurobi_home = "/Library/gurobi951/macos_universal2"
> ```

On first read, this does **not** seem to support 9.5.2 on macOS, since it hard-codes `gurobi951` with no option to change it.

My guess at the best answer would be to glob for `gurobi95*` to find all `9.5.x` versions, and then choose the one with the largest patch version.

When finding the largest patch version, I think it's fine to ignore the possibility of two-digit patch versions (>= 10).  Gurobi's patch versions typically are <= 3 in practice over the years.

If we go that route, then we should play the same game on Ubuntu where anything in `/opt/gurobi95*/linux64` is acceptable and we choose the newest.  (The GUROBI_HOME would still take precedence.)

**Review by mwoehlke-kitware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/18464)* status: 3 unresolved discussions, needs at least two assigned reviewers (waiting on @jwnimmer-tri)

*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake/18464#-NJW3fbg2pyvq_pJEFql:-NJfhhbw5rcmEnt1PGzY:bg62es4) (no related file):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

The file `drake/cmake/modules/FindGurobi.cmake` looks like it should have been updated as well with revised logic.

I think you need to prepare a suite of CI runs that _only_ have 9.5.2 installed, and prove that this PR passes all of those.
</blockquote></details>

`FindGurobi.cmake` has been updated. Some of this needs to run on CI, as I don't have a local install in `/opt`.

___
___
*[`tools/workspace/gurobi/package-ubuntu.BUILD.bazel.in` line 28 at r1](https://reviewable.io/reviews/robotlocomotion/drake/18464#-NJW29pv6yVwlJJkd-HZ:-NJfinCdDGJ3uWZ2czc7:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/fea2405f39a6e28462cbab41cdeb8df18da710d3/tools/workspace/gurobi/package-ubuntu.BUILD.bazel.in#L28)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

typo

Ditto on the next line as well.
</blockquote></details>

Done.
___
*[`tools/workspace/gurobi/repository.bzl` line 17 at r1](https://reviewable.io/reviews/robotlocomotion/drake/18464#-NJVz_T21gyrsb6WSBio:-NJfhabEDjU1KJPjoseM:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/fea2405f39a6e28462cbab41cdeb8df18da710d3/tools/workspace/gurobi/repository.bzl#L17)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

On first read, this does **not** seem to support 9.5.2 on macOS, since it hard-codes `gurobi951` with no option to change it.

My guess at the best answer would be to glob for `gurobi95*` to find all `9.5.x` versions, and then choose the one with the largest patch version.

When finding the largest patch version, I think it's fine to ignore the possibility of two-digit patch versions (>= 10).  Gurobi's patch versions typically are <= 3 in practice over the years.

If we go that route, then we should play the same game on Ubuntu where anything in `/opt/gurobi95*/linux64` is acceptable and we choose the newest.  (The GUROBI_HOME would still take precedence.)
</blockquote></details>

Done.

**Review by jwnimmer-tri — COMMENTED** [review]
> Reviewed 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/18464)* status: 2 unresolved discussions, needs at least two assigned reviewers (waiting on @mwoehlke-kitware)

___
*[`tools/workspace/gurobi/repository.bzl` line 17 at r2](https://reviewable.io/reviews/robotlocomotion/drake/18464#-NJflvpX6Y2HXwYUm5O-:-NJflvpX6Y2HXwYUm5O0:b-tlnop8) ([raw file](https://github.com/robotlocomotion/drake/blob/e9d2c9a98aa596b9b1378f141b40ec81f6af6d74/tools/workspace/gurobi/repository.bzl#L17)):*
> ```Starlark
> 
>     if best_dir == None:
>         fail("Cannot find gurobi")
> ```

When the directory does not exist, we need to defer the error to build-time instead of load-time.  Therefore, we cannot call `fail()` here.  Instead, we should return a dummy path, e.g., `prefix + "xxx" + postfix`.

**Review by mwoehlke-kitware — COMMENTED** [review]
Yup, wasn't sure if this should be feature or fix. Thanks.

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/18464#-:-NJgL-0eElcLAFw7LvP6:bwq5yn6)* status: 2 unresolved discussions, needs at least two assigned reviewers (waiting on @jwnimmer-tri)

___
*[`tools/workspace/gurobi/repository.bzl` line 17 at r2](https://reviewable.io/reviews/robotlocomotion/drake/18464#-NJflvpX6Y2HXwYUm5O-:-NJgKRK7DxIomkbZv6nM:b70dpkc) ([raw file](https://github.com/robotlocomotion/drake/blob/e9d2c9a98aa596b9b1378f141b40ec81f6af6d74/tools/workspace/gurobi/repository.bzl#L17)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

When the directory does not exist, we need to defer the error to build-time instead of load-time.  Therefore, we cannot call `fail()` here.  Instead, we should return a dummy path, e.g., `prefix + "xxx" + postfix`.
</blockquote></details>

That is going to produce a somewhat confusing error message, "Gurobi 9.5 is not installed at /opt/gurobixxx...". Is that acceptable? Do you prefer literal `xxx`, or maybe `/opt/gurboi-notfound`, or `/opt/gurobi<VER>`? Or should the build-time error be changed to remove mention of the non-existing path?)

**Review by jwnimmer-tri — COMMENTED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/18464)* status: 2 unresolved discussions, needs at least two assigned reviewers (waiting on @mwoehlke-kitware)

*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake/18464#-NJW3fbg2pyvq_pJEFql:-NKsWYMP0lyf6mOGLBbp:b-11rsk) (no related file):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

`FindGurobi.cmake` has been updated. Some of this needs to run on CI, as I don't have a local install in `/opt`.
</blockquote></details>

That's fine.  I just need to see the CI evidence before we close out this discussion thread.

___
___
*[`tools/workspace/gurobi/repository.bzl` line 17 at r2](https://reviewable.io/reviews/robotlocomotion/drake/18464#-NJflvpX6Y2HXwYUm5O-:-NKsWAZJD3J-G0EDl2WV:bf3tfm6) ([raw file](https://github.com/robotlocomotion/drake/blob/e9d2c9a98aa596b9b1378f141b40ec81f6af6d74/tools/workspace/gurobi/repository.bzl#L17)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

That is going to produce a somewhat confusing error message, "Gurobi 9.5 is not installed at /opt/gurobixxx...". Is that acceptable? Do you prefer literal `xxx`, or maybe `/opt/gurboi-notfound`, or `/opt/gurobi<VER>`? Or should the build-time error be changed to remove mention of the non-existing path?)
</blockquote></details>

I think the way here currently ("notfound") is fine.
___
*[`tools/workspace/gurobi/repository.bzl` line 10 at r3](https://reviewable.io/reviews/robotlocomotion/drake/18464#-NKsWOp6B-eliQ3zBwbP:-NKsWOp6B-eliQ3zBwbQ:b-bj4kt8) ([raw file](https://github.com/robotlocomotion/drake/blob/d0a57fc567b85815b0d16da29b8f0fe28be5a5fb/tools/workspace/gurobi/repository.bzl#L10)):*
> ```Starlark
>     best_dir = None
>     best_version = None
>     for d in native.glob(prefix + "*" + postfix):
> ```

There is a typo here failing in CI.  The argument to glob is a sequence of pattern strings, not a string.



_Suggestion:_
```Starlark
[prefix + "*" + postfix]
```

**Review by mwoehlke-kitware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/18464)* status: 1 unresolved discussion, needs at least two assigned reviewers (waiting on @jwnimmer-tri)

___
*[`tools/workspace/gurobi/repository.bzl` line 10 at r3](https://reviewable.io/reviews/robotlocomotion/drake/18464#-NKsWOp6B-eliQ3zBwbP:-NKxjGde8cl1N8Ic6L7H:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/d0a57fc567b85815b0d16da29b8f0fe28be5a5fb/tools/workspace/gurobi/repository.bzl#L10)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

There is a typo here failing in CI.  The argument to glob is a sequence of pattern strings, not a string.
</blockquote></details>

Done.

**Review by jwnimmer-tri — COMMENTED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/18464)* status: 1 unresolved discussion, needs at least two assigned reviewers (waiting on @mwoehlke-kitware)

**Review by jwnimmer-tri — COMMENTED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/18464)* status: 1 unresolved discussion, needs at least two assigned reviewers (waiting on @mwoehlke-kitware)

**Comment by jwnimmer-tri** [comment]
CI is still unhappy here.  Even without a copy of Gurobi available locally, you should still be able to `touch` some empty files as a mock-up of what the Gurobi binaries look like, in order to exercise & debug the Bazel logic (even though the compiler would still eventually be unhappy later on).

**Comment by mwoehlke-kitware** [comment]
CI says "The native module can be accessed only from a BUILD thread". So "my guess at the best answer would be to glob for gurobi95* to find all 9.5.x versions" doesn't seem workable. I took a brief look at trying to delay globbing until the BUILD, and it isn't obvious how to do that, given the need for BUILD to be declarative. Any other ideas?

**Comment by jwnimmer-tri** [comment]
Perhaps pass `repo_ctx` into `_find_latest` and then do something like this:

```py
def _find_latest(repo_ctx, prefix, postfix):
    for patch_version in range(9, -1, -1):
        candidate = repo_ctx.path(prefix + patch_version + postfix)
        if candidate.exists():
            return candidate
    return path(prefix + "-notfound" + postfix)
```

**Comment by jwnimmer-tri** [comment]
Oh duh.  We have https://bazel.build/rules/lib/path#readdir to list a directory.  We could call that and then filter the results ourselves to find the subdirectories that match our required pattern.

**Review by jwnimmer-tri — COMMENTED** [review]
+@SeanCurtis-TRI for platform review per schedule, please.

> Reviewed 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/18464#-:-NLNMVKk7DRKoAlZCRJr:bbnvyd5)* status: 1 unresolved discussion, LGTM missing from assignee SeanCurtis-TRI(platform) (waiting on @mwoehlke-kitware and @SeanCurtis-TRI)

*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake/18464#-NJW3fbg2pyvq_pJEFql:-NLNLqJ85ym0BuEjCDok:b7fdv8u) (no related file):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

That's fine.  I just need to see the CI evidence before we close out this discussion thread.
</blockquote></details>

So the evidence we have so far here is just that the new glob-like logic succeeds when run against our Drake CI with Gurobi 9.5.1 installed.  I'm OK merging this PR on the basis of that evidence; it feels pretty convincing.

However, I've edited the PR overview to say `Towards #14157` now instead of closing the issue.  Once this code merges, we can re-assign the issue to the CI maintainers and ask them to upgrade CI to have only Gurobi 9.5.2 installed.  Once that change passes, that will prove that the issue is fully resolved.  That seems easier that trying to test this PR against a canary image of 9.5.2.

___
___
*[`tools/workspace/gurobi/repository.bzl` line 7 at r6](https://reviewable.io/reviews/robotlocomotion/drake/18464#-NLNL-WY8gL0Wjkrssel:-NLNL-WZ2b6In8D0jRdc:bkwcdqm) ([raw file](https://github.com/robotlocomotion/drake/blob/d5f9fa24107dc4b1bbf41035d1b9c6ba91560ee2/tools/workspace/gurobi/repository.bzl#L7)):*
> ```Starlark
> load("@drake//tools/workspace:os.bzl", "determine_os")
> 
> def _find_latest(repo_ctx, path, prefix, subdir):
> ```

nit These arguments are all similar enough that I think we need a docstring here to help clarify what's going on.  Something like:

path: The parent directory where Gurobi would typically be installed, e.g., `/opt` or `/Library`.

prefix: A directory name pattern for Drake's supported major-minor version of Gurobi.  We will search `path` to find a installed copy of Gurobi.

subdir: The platform-specific subdirectory within the Gurobi release, e.g., `linux64`.

**Review by SeanCurtis-TRI — COMMENTED** [review]
:LGTM: modulo the docstring requested below.

> Reviewed 3 of 6 files at r1, 2 of 4 files at r2, 1 of 1 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/18464#-:-NLNVPk14saW-5_lZ5dV:b-3enpjk)* status: 2 unresolved discussions (waiting on @jwnimmer-tri and @mwoehlke-kitware)

___
*[`-- commits` line 9 at r6](https://reviewable.io/reviews/robotlocomotion/drake/18464#-NLNTeSzEzyx0SzcIsg4):*
BTW I have the vague sense that we should prefer leaving github data out of commit messages. THey're great in the github PR description but less so in commits. However, if @jwnimmer-tri didn't flag this, then I presume that vague sense I have has gone awry. Jeremy, have I misapplied a principle here?
___
*[`tools/workspace/gurobi/repository.bzl` line 7 at r6](https://reviewable.io/reviews/robotlocomotion/drake/18464#-NLNL-WY8gL0Wjkrssel:-NLNUttT2wJYnEXi28l5:b-ljm5ep) ([raw file](https://github.com/robotlocomotion/drake/blob/d5f9fa24107dc4b1bbf41035d1b9c6ba91560ee2/tools/workspace/gurobi/repository.bzl#L7)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

nit These arguments are all similar enough that I think we need a docstring here to help clarify what's going on.  Something like:

path: The parent directory where Gurobi would typically be installed, e.g., `/opt` or `/Library`.

prefix: A directory name pattern for Drake's supported major-minor version of Gurobi.  We will search `path` to find a installed copy of Gurobi.

subdir: The platform-specific subdirectory within the Gurobi release, e.g., `linux64`.
</blockquote></details>

Agreed.

It seems to me that the gurobi release will be in `path/prefix/*/subdir` or some such thing. (I'm not sure what the appropriate syntax to indicate zero or more sub-directories between `prefix` and `subdir`.) But that kind of visible concatenation might also contribute to understanding the relationship.

(I may also be fuzzy on whether we expect `prefix` to be a child directory of `path` or merely a descendant. I've assumed child here.)

**Review by jwnimmer-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/18464)* status: 1 unresolved discussion (waiting on @mwoehlke-kitware)

___
*[`-- commits` line 9 at r6](https://reviewable.io/reviews/robotlocomotion/drake/18464#-NLNTeSzEzyx0SzcIsg4):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

BTW I have the vague sense that we should prefer leaving github data out of commit messages. THey're great in the github PR description but less so in commits. However, if @jwnimmer-tri didn't flag this, then I presume that vague sense I have has gone awry. Jeremy, have I misapplied a principle here?
</blockquote></details>

We don't have an official policy.

As an educational experience, sometimes I do remind co-workers that putting an issue number into a commit message turns into an incredible amount of issue-timeline spam:



However, at the moment it's at the author's discretion how they handle this.  I avoid it myself, but it's not a policy.

**Review by SeanCurtis-TRI — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/18464)* status: 1 unresolved discussion (waiting on @jwnimmer-tri)

___
*[`-- commits` line 9 at r6](https://reviewable.io/reviews/robotlocomotion/drake/18464#-NLNTeSzEzyx0SzcIsg4):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

We don't have an official policy.

As an educational experience, sometimes I do remind co-workers that putting and issue number into a commit message turns into an incredible amount of issue-timeline spam:



However, at the moment it's at the author's discretion how they handle this.  I avoid it myself, but it's not a policy.
</blockquote></details>

Thanks

**Review by mwoehlke-kitware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/18464)* status: :shipit: complete! all discussions resolved, LGTM from assignees jwnimmer-tri(platform),SeanCurtis-TRI(platform) (waiting on @jwnimmer-tri and @SeanCurtis-TRI)

___
*[`tools/workspace/gurobi/repository.bzl` line 7 at r6](https://reviewable.io/reviews/robotlocomotion/drake/18464#-NLNL-WY8gL0Wjkrssel:-NLRLo_U8C9UgxaWQXok:ba3t6xd) ([raw file](https://github.com/robotlocomotion/drake/blob/d5f9fa24107dc4b1bbf41035d1b9c6ba91560ee2/tools/workspace/gurobi/repository.bzl#L7)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

Agreed.

It seems to me that the gurobi release will be in `path/prefix/*/subdir` or some such thing. (I'm not sure what the appropriate syntax to indicate zero or more sub-directories between `prefix` and `subdir`.) But that kind of visible concatenation might also contribute to understanding the relationship.

(I may also be fuzzy on whether we expect `prefix` to be a child directory of `path` or merely a descendant. I've assumed child here.)
</blockquote></details>

It's equivalent to globbing for `f'{path}/{prefix}*/{subdir}'` :slightly_smiling_face:. I added a note to that effect.

**Review by jwnimmer-tri — COMMENTED** [review]
> Reviewed 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/18464)* status: :shipit: complete! all discussions resolved, LGTM from assignees jwnimmer-tri(platform),SeanCurtis-TRI(platform) (waiting on @mwoehlke-kitware)

**Review by SeanCurtis-TRI — COMMENTED** [review]
> Reviewed 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/18464)* status: :shipit: complete! all discussions resolved, LGTM from assignees jwnimmer-tri(platform),SeanCurtis-TRI(platform) (waiting on @mwoehlke-kitware)

**Comment by jwnimmer-tri** [comment]
Oops.  All three of us forgot to test on macOS.  Hotfixed in #18563.
